src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,68 @@
+import classNames from 'classnames';
+import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  index: number;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  index,
+}) => {
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  const selectTodoHandler = (currentTodo: Todo) => {
+    dispatch(actions.setTodo(currentTodo));
+  };
+
+  return (
+    <tr
+      data-cy="todo"
+      className={classNames({
+        'has-background-info-light': selectedTodo?.id === todo.id,
+      })}
+    >
+      <td className="is-vcentered">{index + 1}</td>
+
+      <td className="is-vcentered">
+        {todo.completed && (
+          <span className="icon" data-cy="iconCompleted">
+            <i className="fas fa-check" />
+          </span>
+        )}
+      </td>
+
+      <td className="is-vcentered is-expanded">
+        <p className={classNames({
+          'has-text-danger': !todo.completed,
+          'has-text-success': todo.completed,
+        })}
+        >
+          {todo.title}
+        </p>
+      </td>
+
+      <td className="has-text-right is-vcentered">
+        <button
+          data-cy="selectButton"
+          className="button"
+          type="button"
+          onClick={() => selectTodoHandler(todo)}
```

Comment:
Avoid creating new functions within the render method. This could lead to unnecessary re-renders as a new function is created every time component re-renders.